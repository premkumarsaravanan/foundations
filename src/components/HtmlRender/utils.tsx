import * as React from 'react'
import { H4, H5 } from '../Typography'

export interface Element {
  type: string
  tagName?: string
  children?: Element[]
  attributes?: any
  content?: string
}

const sortContentType = (domItem: Element, index: number) =>
  domItem.type === 'text' ? domItem.content || null : rendererModule.sortTags(domItem, index)

const getChildren = (domTag: Element) =>
  domTag.children
    ? domTag.children.map((child: Element, index: number) => rendererModule.sortContentType(child, index))
    : null

const getAttributes = (domTag: Element, index: number) => {
  const attributes = domTag.attributes || {}
  return { ...attributes, style: {}, key: index }
}

const sortTags = (domTag: Element, index: number) => {
  const children = rendererModule.getChildren(domTag)
  const attributes = rendererModule.getAttributes(domTag, index)
  if (!children || !children.length) {
    return null
  }
  switch (domTag.tagName) {
    case 'p':
      return <p {...attributes}>{children}</p>
    case 'a':
      return <a {...attributes}>{children}</a>
    case 'b':
      return <b {...attributes}>{children}</b>
    case 'u':
      return <u {...attributes}>{children}</u>
    case 'ul':
      return <ul {...attributes}>{children}</ul>
    case 'ol':
      return <ol {...attributes}>{children}</ol>
    case 'li':
      return <li {...attributes}>{children}</li>
    case 'h1':
      return <H4 {...attributes}>{children}</H4>
    case 'h2':
      return <H5 {...attributes}>{children}</H5>
    case 'i':
      return <i {...attributes}>{children}</i>
    case 'strike':
      return <del {...attributes}>{children}</del>
    case 'blockquote':
      return <blockquote {...attributes}>{children}</blockquote>
    case 'pre':
      return <pre {...attributes}>{children}</pre>
    case 'hr':
      return <hr {...attributes} />
    default:
      return <div {...attributes}>{children}</div>
  }
}

const renderer = (domContent: Element[]) =>
  domContent.map((domItem: Element, index: number) => rendererModule.sortContentType(domItem, index))

export const rendererModule = {
  sortContentType,
  getChildren,
  getAttributes,
  sortTags,
  renderer
}
export default renderer
