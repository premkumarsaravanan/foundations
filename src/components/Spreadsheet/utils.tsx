import * as React from 'react'
import Papa from 'papaparse'
import { Cell } from './types'

/** Get max row and col of data */
export const getMaxRowAndCol = (data: Cell[][]): { maxRow: number; maxCol: number } => {
  const maxRow = data.length
  /* default to 0 */
  let maxCol = 0
  /* check every row to find max length of column */
  data.forEach(row => {
    const numberOfCurrentRowColumn = row.length
    if (maxCol < numberOfCurrentRowColumn) {
      maxCol = numberOfCurrentRowColumn
    }
  })
  return { maxRow, maxCol }
}

export const setCurrentCellValue = (
  cellData: string,
  data: Cell[][],
  row: number,
  col: number,
  setData: React.Dispatch<Cell[][]>
): void => {
  const newData = JSON.parse(JSON.stringify(data))
  newData[row][col].value = cellData
  setData(newData)
}

export const parseCsvFile = (file: File): Promise<Papa.ParseResult> =>
  new Promise(resolve => {
    Papa.parse(file, {
      complete: (results: Papa.ParseResult) => {
        resolve(results)
      }
    })
  })

/** convert to string to download */
export const unparseDataToCsvString = (parseResult: string[][]): string => Papa.unparse(parseResult)
/**
 * Convert from ParseResult to Cell[][]
 * Convert from [['value1', 'value2'], ['value3', 'value4']]
 * to [[{value: 'value1'}, {value: ''}], [{value: 'value3'}, {value: 'value4'}]]
 * for compatible reason
 */
export const convertToCompatibleData = (parsedResult): Cell[][] =>
  parsedResult.data.map(rowArray => rowArray.map(value => ({ value })))

/**
 * Convert back from Cell[][] to string[][]
 */
export const convertDataToCsv = (data: Cell[][]): string[][] => data.map(rowArray => rowArray.map(({ value }) => value))
