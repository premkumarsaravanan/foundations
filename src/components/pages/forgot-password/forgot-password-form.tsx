import React from 'react'
import { Form, withFormik, FormikProps, FormikBag, FormikErrors } from 'formik'
import { compose } from 'redux'
import { FlexContainerResponsive, GridItem, FormSubHeading, Grid, Input, Button } from '@reapit/elements'
import { RouterProps, withRouter } from 'react-router'
import Routes from '@/constants/routes'

export type ForgotPasswordFormProps = FormikProps<ForgotPasswordValues>

export const ForgotPasswordForm: React.FC<ForgotPasswordFormProps> = ({ isSubmitting, isValidating, isValid }) => {
  return (
    <Form>
      <FormSubHeading>Please enter your email address to reset your password</FormSubHeading>
      <Grid>
        <GridItem>
          <Input dataTest="email" type="email" labelText="Email" id="email" name="email" />
        </GridItem>
      </Grid>
      <FlexContainerResponsive>
        <Button disabled={!isValid} loading={isSubmitting || isValidating} variant="primary" type="submit">
          Submit
        </Button>
      </FlexContainerResponsive>
    </Form>
  )
}

export type ForgotPasswordValues = {
  email: string
}

export const mapPropsToValues = (): ForgotPasswordValues => ({
  email: ''
})

export const requestForgotPassword = (values: ForgotPasswordValues) => {
  console.log(values)
  // TODO: will replace mock here for real API
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve(true)
    }, 500)
    setTimeout(() => {
      reject(new Error('some errors'))
    }, 500)
  })
}

export const handleSubmitForgotPassword = async (
  values: ForgotPasswordValues,
  { setSubmitting, props, setErrors }: FormikBag<EnhanceForgotPasswordFormProps & RouterProps, ForgotPasswordValues>
) => {
  try {
    const response = await requestForgotPassword(values)
    if (response) {
      setSubmitting(false)
      props.history.push(`${Routes.FORGOT_PASSWORD}?isSuccessRequestResetPassword=1`)
    }
  } catch (error) {
    console.error(error)
    setErrors({ email: 'We could not find an account registered with that email address' })
    setSubmitting(false)
  }
}

export const withForgotPasswordForm = withFormik({
  displayName: 'WithForgotPasswordForm',
  mapPropsToValues,
  handleSubmit: handleSubmitForgotPassword
})

export type EnhanceForgotPasswordFormProps = {}

const EnhanceForgotPasswordForm = compose<React.FC<EnhanceForgotPasswordFormProps>>(
  withRouter,
  withForgotPasswordForm
)(ForgotPasswordForm)
EnhanceForgotPasswordForm.displayName = 'EnhanceForgotPasswordForm'

export default EnhanceForgotPasswordForm
