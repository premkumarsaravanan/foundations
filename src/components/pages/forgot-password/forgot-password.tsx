import React from 'react'
import { H1 } from '@reapit/elements'
import { withRouter, RouteComponentProps } from 'react-router'
import { compose } from 'redux'
import logoImage from '@/assets/images/reapit-graphic.jpg'
import loginStyles from '@/styles/pages/login.scss?mod'
import EnhanceForgotPasswordForm from './forgot-password-form'

export type ForgotPasswordProps = {} & RouteComponentProps

export const ForgotPassword: React.FC<ForgotPasswordProps> = ({ location }) => {
  const isSuccessRequestResetPassword = location?.search === '?isSuccessRequestResetPassword=1'
  return (
    <div className={loginStyles.container}>
      <div className={`${loginStyles.wrapper} ${loginStyles.isSubmitting && loginStyles.disabled}`}>
        <H1 isCentered>Forgot password</H1>
        <p className="pb-8">Welcome to Reapit Foundations</p>
        {isSuccessRequestResetPassword ? (
          <p className="pb-8">Please check your email inbox for the next step to reset your password</p>
        ) : (
          <EnhanceForgotPasswordForm />
        )}
      </div>
      <div className={loginStyles.image}>
        <img src={logoImage} alt="Reapit Graphic" />
      </div>
    </div>
  )
}

export default compose<React.FC<ForgotPasswordProps>>(withRouter)(ForgotPassword)
