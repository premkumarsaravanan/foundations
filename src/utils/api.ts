import { getAccessToken } from './session'
import { CONTACTS_HEADERS } from '../constants/api'

export const initAuthorizedRequestHeaders = async () => ({
  ...CONTACTS_HEADERS,
  Authorization: `Bearer ${await getAccessToken()}`
})
