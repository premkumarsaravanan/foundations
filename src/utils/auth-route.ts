import { LoginType } from '@reapit/cognito-auth'
import Routes from '../constants/routes'

export function getAuthRouteByLoginType(loginType: LoginType) {
  switch (loginType) {
    case 'ADMIN':
      return Routes.ADMIN_LOGIN
    case 'DEVELOPER':
      return Routes.DEVELOPER_LOGIN
    default:
      return Routes.CLIENT_LOGIN
  }
}

export function getLoginTypeByPath(path: string) {
  switch (path) {
    case Routes.ADMIN_LOGIN:
      return 'ADMIN'
      break
    case Routes.DEVELOPER_LOGIN:
      return 'DEVELOPER'
      break
    default:
      return 'CLIENT'
  }
}
