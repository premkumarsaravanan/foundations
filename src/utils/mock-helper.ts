import { RouteComponentProps } from 'react-router'
import { Href } from 'history'

export function getMockRouterProps<P>(data: P) {
  const location = {
    hash: '',
    key: '',
    pathname: '',
    search: '?token=123',
    state: {}
  }

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params: data,
      path: '',
      url: ''
    },
    location: location,
    history: {
      length: 2,
      action: 'POP',
      location: location,
      push: jest.fn(),
      replace: jest.fn(),
      go: jest.fn(),
      goBack: jest.fn(),
      goForward: jest.fn(),
      block: jest.fn(),
      createHref: () => '',
      listen: jest.fn()
    },
    staticContext: {}
  }

  return props
}
