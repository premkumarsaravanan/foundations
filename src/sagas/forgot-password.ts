import { fetcher } from '@reapit/elements'
import { MARKETPLACE_HEADERS } from '../constants/api'
import { put, fork, takeLatest, all, call } from '@redux-saga/core/effects'
import ActionTypes from '../constants/action-types'
import { history } from '../core/router'
import Routes from '@/constants/routes'
import { Action } from '@/types/core'
import { forgotPasswordLoading } from '@/actions/forgot-password'

export const callAPIForgotPassword = async email => {
  const FORGOT_PASSWORD_URL = '/password/reset'
  const response = await fetcher({
    url: FORGOT_PASSWORD_URL,
    // please refer to this ticket https://reapit.atlassian.net/browse/CLD-620
    api: 'https://rbsbshnxvb.execute-api.eu-west-2.amazonaws.com/dev/api',
    method: 'POST',
    headers: MARKETPLACE_HEADERS,
    body: {
      userName: email
    }
  })
  return response
}

export const requestForgotPassword = function*({ data: email }) {
  yield put(forgotPasswordLoading(true))
  try {
    const response = yield call(callAPIForgotPassword, email)
    if (response.CodeDeliveryDetails) {
      yield history.push(`${Routes.FORGOT_PASSWORD}?isSuccess=1`)
    }
  } catch (error) {
    console.error(error.message)
    yield history.push(`${Routes.FORGOT_PASSWORD}?isError=1`)
  } finally {
    yield put(forgotPasswordLoading(false))
  }
}

export const requestForgotPasswordListen = function*() {
  yield takeLatest<Action<string>>(ActionTypes.FORGOT_PASSWORD_SUBMIT_EMAIL, requestForgotPassword)
}

const forgotPasswordSagas = function*() {
  yield all([fork(requestForgotPasswordListen)])
}

export default forgotPasswordSagas
