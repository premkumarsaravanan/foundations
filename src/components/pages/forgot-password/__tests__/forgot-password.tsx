import React from 'react'
import { shallow } from 'enzyme'
import { ForgotPassword } from '../forgot-password'
import { getMockRouterProps } from '@/utils/mock-helper'

describe('ForgotPassword', () => {
  it('should match snapshot', () => {
    const mockProps = {
      ...getMockRouterProps({}),
      location: {
        hash: '',
        key: '',
        pathname: '',
        state: {},
        search: '?isSuccessRequestResetPassword=1'
      }
    }
    const wrapper = shallow(<ForgotPassword {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('[className="pb-8"]')).toHaveLength(2)
  })

  it('should match snapshot', () => {
    const mockProps = {
      ...getMockRouterProps({}),
      location: {
        hash: '',
        key: '',
        pathname: '',
        state: {},
        search: ''
      }
    }
    const wrapper = shallow(<ForgotPassword {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('EnhanceForgotPasswordForm')).toHaveLength(1)
  })
})
