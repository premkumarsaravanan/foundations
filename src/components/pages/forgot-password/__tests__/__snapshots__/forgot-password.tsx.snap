// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword should match snapshot 1`] = `
<div>
  <div
    className="undefined undefined"
  >
    <H1
      isCentered={true}
    >
      Forgot password
    </H1>
    <p
      className="pb-8"
    >
      Welcome to Reapit Foundations
    </p>
    <p
      className="pb-8"
    >
      Please check your email inbox for the next step to reset your password
    </p>
  </div>
  <div>
    <img
      alt="Reapit Graphic"
      src=""
    />
  </div>
</div>
`;

exports[`ForgotPassword should match snapshot 2`] = `
<div>
  <div
    className="undefined undefined"
  >
    <H1
      isCentered={true}
    >
      Forgot password
    </H1>
    <p
      className="pb-8"
    >
      Welcome to Reapit Foundations
    </p>
    <EnhanceForgotPasswordForm />
  </div>
  <div>
    <img
      alt="Reapit Graphic"
      src=""
    />
  </div>
</div>
`;
